### ISSUE TITLE
advice hyper-parameter setting on large KG 

### ISSUE THREAD

**Issue by walton-wang929** [awslabs/dgl-ke#163]
hello guys, thanks for your great work.  **do you have any suggestions on the hyper-parameter setting?** I didn't find any details on both paper[**_2020-SIGAR-DGL-KE: Training Knowledge Graph Embeddings at Scale_**] and this git. 

I am training a large KG, here are the KG intro:

The total of relations is 1825826.
The total of entities is 76914149.
The total of train triples is 382606627.
The total of test triples is 58387397.
The total of valid triples is 58387435.

so far, due to the CPU memory limit. I tried this configuration shell:
`DGLBACKEND=pytorch dglke_train --model_name TransE_l2 --data_path ./data/360KG/ --format udd_hrt --dataset 360KG --data_files entities.dict relation.dict train.txt valid.txt test.txt --save_path ./run-exp/360KG --max_step 30000 --batch_size 1000 --batch_size_eval 16 --neg_sample_size 200 --log_interval 100 --hidden_dim 64 --gamma 19.9 --lr 0.25 --regularization_coef 1.00E-09 --test  -adv --mix_cpu_gpu --num_proc 40 --num_thread 20 --force_sync_interval 1000 --gpu 0 1 2 3 --regularization_coef 1e-9 --neg_sample_size_eval 10000 --no_eval_filter`

the results are below:
training takes 7777.839455842972 seconds
Save model to ./run-exp/360KG/TransE_l2_360KG_2

Test average MRR : 0.2841196238250153
Test average MR : 2433.0077265990762
Test average HITS@1 : 0.23738878507272357
Test average HITS@3 : 0.30359764631394076
Test average HITS@10 : 0.37124394107248765
testing takes 16571.815 seconds

the MRR and Hits@10 are very low. so guys, very appreciate your suggestions on the hyper-parameter setting.

**Comment by classicsong** [comment]
Usually, several things to tune, 1) gamma, this impact how data is initialized and the score func; 2) You can change your dim to 128 or 256, 3) learning rate as well as max steps.
You can look at the config in https://github.com/awslabs/dgl-ke/blob/master/examples/freebase/multi_gpu.sh

**Comment by walton-wang929** [comment]
very clearly. let me try ha. thanks.

**Comment by walton-wang929** [comment]
for a large graph, data loading are very slow, which parameter can boost data loading ?
